### PR TITLE
Use POST for retrieving STS credentials from AWS

### PR DESCRIPTION
**Pull request by lnattrass**
This will prevent the token from being leaked in the query string when the request fails.

### CODE REVIEW

File: plugins/rest/aws.go [hunk 1]
```diff
@@ -317,11 +317,11 @@ func (cs *awsWebIdentityCredentialService) refreshFromService() error {
...
 	}
 	stsRequestURL, _ := url.Parse(cs.stsPath())
-	stsRequestURL.RawQuery = queryVals.Encode()
 
 	// construct an HTTP client with a reasonably short timeout
 	client := &http.Client{Timeout: time.Second * 10}
-	req, err := http.NewRequest(http.MethodGet, stsRequestURL.String(), nil)
+	req, err := http.NewRequest(http.MethodPost, stsRequestURL.String(), strings.NewReader(queryVals.Encode()))
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
 	if err != nil {
```
Comment (srenatus) [inline plugins/rest/aws.go:325]: Can we do that after the err check? The chances are slim, but better safe than sorry...

**Review by srenatus — COMMENTED** [review]
Looks good,one nitpick

File: plugins/rest/aws.go (hunk 1 above)
Comment (lnattrass) [reply plugins/rest/aws.go:325]: Ah yep, totally!

**Review by tsandall — APPROVED** [review]
LGTM. @lnattrass have you tested this manually?

**Comment by lnattrass** [comment]
Confirming that this has been tested manually and functions correctly:

```
{"level":"error","msg":"Bundle load failed: request failed: error getting AWS credentials: STS HTTP request failed: Post \"https://sts.us-west-2.amazonaws.com\": dial tcp 10.6.1.87:443: connect: connection refused","name":"service","plugin":"bundle","time":"2021-07-11T15:20:19Z"}
```

**Comment by tsandall** [comment]
Cool @lnattrass ! Just to confirm that you tested the positive case as well (i.e., the error message looks good, I just want to make sure we don't break users on this one since we don't have any system test coverage unfortunately.)

**Comment by lnattrass** [comment]
Yes, this went healthy when STS connectivity was restored.

**Comment by tsandall** [comment]
Perfect. Thanks!

**Pull request merged**